### PR TITLE
Deal with apple LTR longtail phantom data

### LINKED ISSUE #112: apple LTR logtail data
apple LTR histogram data can contain a quite long logtail which is super low, e.g. data of 1h to the length of the episode of just one listener.

Example of an 1 hour episode:

```
{"0": 1659},
{"15": 1468}...{"3375": 1106},
{"3390": 1052},
{"3405": 1018},
{"3420": 482},
{"3435": 377},
{"3450": 236},
{"3465": 182},
{"3480": 120},
{"3495": 89},
{"3510": 67},
{"3525": 48},
{"3540": 28},
{"3555": 20},
{"3570": 8},
{"3585": 3},
{"3600": 2},
{"3615": 2},
{"3630": 1},
{"3645": 1},
{"3660": 1},
{"3675": 1},
{"3690": 1},
{"3705": 1},
{"3720": 1},
{"3735": 1},
{"3750": 1},
{"3765": 1},
{"3780": 1},
{"3795": 1},
{"3810": 1},
{"3825": 1},
{"3840": 1}...{"8625": 1} 
```

### PR DESCRIPTION
**Pull request by woolfg**
fix #112

### CODE REVIEW

File: src/stats/performance.ts [hunk 1]
```diff
@@ -27,10 +27,22 @@ const calcApplePodcastPerformanceQuarters = function (
 
     const maxListeners = Math.max(...performanceValues)
 
-    if (performanceValues.length < 4) {
-        throw Error('too little data to calculate quarters')
+    // apple LTR data can contain q weird long tail which is a way longer than the episode itself
```
Comment (mre) [inline src/stats/performance.ts:30]:
```suggestion
    // Apple LTR data can contain a weird long tail, which is a way longer than the episode itself
```

File: src/stats/performance.ts [hunk 2]
```diff
@@ -27,10 +27,22 @@ const calcApplePodcastPerformanceQuarters = function (
...
+    const longtailThresholdValue = maxListeners * longtailThreshold
+    let longtailIndex = performanceValues.length - 1
+    while (
+        longtailIndex > 0 &&
+        performanceValues[longtailIndex] <= longtailThresholdValue
+    ) {
+        longtailIndex--
     }
+    // remove longtail from the data array
+    performanceValues.splice(longtailIndex + 1)
```
Comment (mre) [inline src/stats/performance.ts:43]: We could put that part into a separate function to make testing easier. Let's maybe add a few tests for the trimming algorithm, like `[], [1], [-1], [0.1, 1000], [1000, 0.01, 4000], [1000, 0.1, 0.1, 0.1]`

Comment (mre) [reply src/stats/performance.ts:43]: something like `cutAtThreshold(series, threshold)`

File: src/stats/performance.test.ts [hunk 3]
```diff
@@ -25,18 +26,56 @@ describe('calculate quarterly performance data of an episode', () => {
...
+        ])
+        // expect a cut at 5% which is 231*0.05 = 11.55
+        const res = calcApplePodcastPerformanceQuarters(testdata)
+        expect(res.maxListeners).toBe(300)
+        expect(res.quarterMedianValues).toStrictEqual([225, 205, 45, 16])
+    })
+})
+
+describe('remove longtail from performance data', () => {
+    it('simple case', () => {
```
Comment (mre) [inline src/stats/performance.test.ts:64]: While we're at it, let's add a few pathological cases. 😝 

```suggestion
   // Test with empty data
    it('empty data', () => {
        const testdata = []
        const res = removeLongtailFromPerformanceData(testdata, 0.05)
        expect(res.maxListeners).toBeUndefined()
        expect(res.performanceValues).toStrictEqual([])
    })

    // Test with only one item
    it('single item data', () => {
        const testdata = [1]
        const res = removeLongtailFromPerformanceData(testdata, 0.05)
        expect(res.maxListeners).toBe(1)
        expect(res.performanceValues).toStrictEqual([1])
    })

    // Test with negative numbers
    it('negative numbers', () => {
        const testdata = [1, 2, 3, -4, -5]
        const res = removeLongtailFromPerformanceData(testdata, 0.05)
        expect(res.maxListeners).toBe(3)
        expect(res.performanceValues.every((v) => v > 3 * 0.05)).toBe(true)
    })

    // Test with zeros
    it('zeroes in data', () => {
        const testdata = [0, 0, 0, 0, 0, 0]
        const res = removeLongtailFromPerformanceData(testdata, 0.05)
        expect(res.maxListeners).toBe(0)
        expect(res.performanceValues).toStrictEqual([0, 0, 0, 0, 0, 0])
    })

    // Test with all identical values
    it('all identical values', () => {
        const testdata = [5, 5, 5, 5, 5]
        const res = removeLongtailFromPerformanceData(testdata, 0.05)
        expect(res.maxListeners).toBe(5)
        expect(res.performanceValues).toStrictEqual([5, 5, 5, 5, 5])
    })
    
    it('simple case', () => {
```

Comment (woolfg) [reply src/stats/performance.test.ts:64]: thanks, added some useful and defined tests and skipped the not defines/useless ones to dave some CO2 ;)

**Review by mre — APPROVED** [review]
lgtm

**Pull request merged**